src/containerapp/azext_containerapp/_utils.py

```diff
@@ -1441,10 +1451,186 @@ def set_managed_identity(cmd, resource_group_name, containerapp_def, system_assi
                 containerapp_def["identity"]["userAssignedIdentities"][r] = {}
 
 
-def _validate_custom_loc_and_location(cmd, custom_location):
-    from ._client_factory import customlocation_client_factory
-    parsed_custom_loc = parse_resource_id(custom_location)
+def _validate_custom_loc_and_location(cmd, custom_location=None, env=None, connected_cluster_id=None, env_rg=None):
+    from ._up_utils import list_connected_environments
+
+    if not is_valid_resource_id(custom_location):
+        raise ValidationError('{} is not a valid Azure resource ID.'.format(custom_location))
+
+    try:
+        r = get_custom_location(cmd=cmd, custom_location_id=custom_location)
+    except:
+        raise ResourceNotFoundError("Cannot find custom location with custom location ID {}".format(custom_location))
+
+    if connected_cluster_id:
+        if connected_cluster_id.lower() != r.host_resource_id.lower():
+            raise ValidationError('Custom location {} not in cluster {}.'.format(custom_location, connected_cluster_id))
+
+    # check env
+    if env:
+        env_list = [e for e in
+                    list_connected_environments(cmd=cmd, resource_group_name=env_rg, custom_location=custom_location) if
+                    (e["id"].lower() != env.lower() and e["name"] != env)]
+        if len(env_list) > 0:
+            raise ValidationError('The provided custom location already used by other environment. If you want to use this custom location, please specify associated environment with --environment. \n Otherwise, please use another custom location.')
+
+    # check extension type
+    check_extension_type = False
+    for extension_id in r.cluster_extension_ids:
+        extension = get_cluster_extension(cmd, extension_id)
+        if extension.extension_type.lower() == CONTAINER_APP_EXTENSION_TYPE:
+            check_extension_type = True
+            break
+    if not check_extension_type:
+        raise ValidationError('There is no Microsoft.App.Environment extension found associated with custom location {}'.format(custom_location))
+
+    return r.location
```

Comment:
clarify what you do here and what is the difference with _get_custom_location_and_extension_id_and_location_from_cluster

Reply:
done